ts/src/grpc.ts

```diff
@@ -102,17 +124,46 @@ export namespace grpc {
     return new Uint8Array(frame);
   }
 
-  function getStatusFromHeaders(headers: BrowserHeaders): Code {
+  function getStatusFromHeaders(headers: BrowserHeaders): Code | null {
     const fromHeaders = headers.get("grpc-status") || [];
     if (fromHeaders.length > 0) {
       try {
         const asString = fromHeaders[0];
         return parseInt(asString, 10);
       } catch (e) {
-        return Code.Internal;
+        return null;
       }
     }
-    return Code.Internal;
+    return null;
+  }
+
+  export function unary<TRequest extends jspb.Message, TResponse extends jspb.Message, M extends UnaryMethodDefinition<TRequest, TResponse>>(methodDescriptor: M,
```

Comment:
What happens if the client makes a `unary()` call against a streaming endpoint?  If the result is predictable (ie: just the first message in the stream) then that would make sense, however if it's not predictable then I would suggest we throw a runtime exception to avoid suprises.